packages/core/test/specHelper.ts

```diff
@@ -417,3 +424,80 @@ export function stubCookie() {
     },
   }
 }
+
+export function stubReportingObserver() {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** Maybe we'll need to split this file one day, it's beginning to be a bit large

Reply:
I agree